CMakeLists.txt

```diff
@@ -12,6 +12,30 @@
 #=========================================================================================
 
 cmake_minimum_required(VERSION 3.12)
+
+# Load machine specific defaults such as architecture and mpi launch command.
+# Command line argument takes precedence over environment variable.
+# Loading this before project definition to allow setting the compiler.
+if (MACHINE_CFG)
+  if(EXISTS "${MACHINE_CFG}")
+    include(${MACHINE_CFG})
+  else()
+    message(FATAL_ERROR "Given machine configuration at "
+      "${MACHINE_CFG} not found.")
+  endif()
+elseif (DEFINED ENV{MACHINE_CFG})
+  if(EXISTS "$ENV{MACHINE_CFG}")
+    include($ENV{MACHINE_CFG})
+  else()
+    message(FATAL_ERROR "Given machine configuration from environment variable "
+      "MACHINE_CFG at $ENV{MACHINE_CFG} not found.")
+  endif()
+else()
+  message(WARNING "Not using any machine configuration. Consider creating a configuration "
+    "file following the examples in ${PROJECT_SOURCE_DIR}/cmake/machine_cfgs/ and then "
+    "point the MACHINE_CFG variable to your custom file.")
```

Comment:
I also like this warning message.

Reply:
I'd make a point, I suppose, that these machine files can live outside the repo for user-configurations.

Reply:
I added a note on that.